Core/KSP.cs

```diff
@@ -321,9 +321,16 @@ public string CkanDir()
 
         public string DownloadCacheDir()
         {
-            return KSPPathUtils.NormalizePath(
-                Path.Combine(CkanDir(), "downloads")
-            );
+            var registry = RegistryManager.Instance(this).registry;
```

Comment:
This causes all of the instances of the registry to be loaded at start. Since a current registry is 20 MB, this would be quite a delay on systems with many instances. It also causes an unworkaroundable exception if you have an instance with a corrupted registry.json (as I do).